### PR TITLE
Initial configuration

### PR DESCRIPTION
**Pull request by cinkonaap**
I've provided initial setup that works in our application.

For time being the cli script always exit with success code, so we will have time to fix them and agree on rules before it fail our builds.

After POC with @jedrula, proposed rule-set is:

```
'block-indentation': 2
'html-comments': true
'nested-interactive': true
'self-closing-void-elements': true
'triple-curlies': true
'deprecated-each-syntax': true
'link-rel-noopener': true
'invalid-interactive': true
'img-alt-attributes': true
'style-concatenation': true
'deprecated-inline-view-helper': true
'unused-block-params': true
'bare-strings': true
'inline-link-to': true
```

plus in the future:
```
'inline-styles': true // not working for now
```

### CODE REVIEW

File: bin/hbslint.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+// prepare linter service
+let config = {};
+Object.assign(config, getConfigPath(argv));
+const linter = new Linter(config);
+
+// prepare lookup
+const lintDirectories = argv['_'];
+
+const dirsToTemplates = lintDirectories.map((directory) => {
```
Comment (jedrula) [inline bin/hbslint.js]: I would consider building `templates` right away with a reduce.

File: bin/hbslint.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  });
+
+  if (errors.length) {
+    errorFilesCount++;
+
+    if (isVerbose) {
+      log(`\n${fullPath}`);
+
+      errors.forEach(error => {
+        const { line, column, rule, message } = error;
```
Comment (jedrula) [inline bin/hbslint.js]: I am not sure if we shouldn't stick to the oldschool js - no arrow functions and destrucuring. If this was to make it's way into ember-cli I think it needs to run on node versions that do not support these features

File: bin/hbslint.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+        // TODO use join here
```
Comment (jedrula) [inline bin/hbslint.js]: what do you mean by join?

File: bin/hbslint.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+  // TODO change this to exitCode 1 after we fix most of bugs
+  exitCode = 0;
+} else {
+  log(chalk.green('No warnings found'));
+}
+
+process.on("exit", function() {
+  process.exit(exitCode);
+});
```
Comment (jedrula) [inline bin/hbslint.js]: I think everything is synchronous here so we could probably just do `process.exit(exitCode);` without the `on exit`

File: bin/hbslint.js (hunk 3 above)
Comment (cinkonaap) [inline bin/hbslint.js]: I would create array with data, and to `Array.prototype.join(' ')` to create this spaces @jedrula

**Comment by cinkonaap** [comment]
@jedrula Please take another look 👀

File: README.md [hunk 5]
```diff
@@ -1 +1,33 @@
...
+npm package with AirHelp Handlebars Lint shareable cli and config
+
+## Usage
+This package provides cli to lint your project against config. It also provides config used in AirHelp Ember applications.
+
+### cli
+To run linter in your build environment, create npm script that runs provided `hbslint` cli against your project source.
+
+1. `hbslint app/templates app/components`
+2. Use `verbose` option to see check errors
```
Comment (jedrula) [inline README.md]: to see or to check ? :)

File: bin/hbslint.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  const dirTemplates = walkSync(directory)
+    .filter(file => path.extname(file) === '.hbs')
+    .map(file => path.join(directory, file));
+
+  templates.push(...dirTemplates);
+
+  return templates;
+}, []);
+
+const templates = [].concat.apply([], dirsToTemplates);
```
Comment (jedrula) [inline bin/hbslint.js]: is this line necessary ? I thought that the reduce will produce the final flat array ?

**Comment by jedrula** [comment]
looks usable to me 👍

**Comment by springerigor** [comment]
I assume that dependencies can be moved to dev dependencies, other than that good job 👍

**Pull request merged**